tensorpipe/test/channel/channel_test.h

```diff
@@ -20,71 +20,168 @@
 #include <tensorpipe/test/peer_group.h>
 #include <tensorpipe/transport/uv/context.h>
 
-class DataWrapper {
+template <typename TBuffer>
+class DataWrapper {};
+
+template <>
+class DataWrapper<tensorpipe::CpuBuffer> {
  public:
-  virtual ~DataWrapper() = default;
-  virtual void* data() = 0;
-  virtual size_t size() = 0;
-  virtual void wrap(const void* ptr) = 0;
-  virtual void unwrap(void* ptr) = 0;
+  explicit DataWrapper(size_t length) : vector_(length) {}
+
+  explicit DataWrapper(const std::vector<uint8_t>& v) : vector_(v) {}
+
+  tensorpipe::CpuBuffer buffer() const {
+    return tensorpipe::CpuBuffer{
+        .ptr = const_cast<uint8_t*>(vector_.data()),
+        .length = vector_.size(),
```

Comment:
Didn't you say that this is a C++-20 thing?

Reply:
This is my understanding of https://en.cppreference.com/w/cpp/language/aggregate_initialization. Basically it's a C99 thing that hasn't made it in the C++ standard before C++20, but most compilers accept it IIUC. Anyways, reverting to something officially supported.